backend/analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/container/analysis/filesystem/FileMetadata.java

```diff
@@ -161,10 +161,13 @@ public Metadata readMetadata(InputStream stream, Charset encoding, URI fileUri,
 
   private static InputStream streamFile(File file) {
     try {
-      return new BOMInputStream(new FileInputStream(file),
-        ByteOrderMark.UTF_8, ByteOrderMark.UTF_16LE, ByteOrderMark.UTF_16BE, ByteOrderMark.UTF_32LE, ByteOrderMark.UTF_32BE);
+      return BOMInputStream.builder().setInputStream(new FileInputStream(file))
```

Comment:
I have the impression this `streamFile` is dead code. Maybe we could instead remove the method?

Reply:
Good catch! I'll remove the code and attach the ticket to the related Jira ticket. Thanks